site/content/docs/developer-guide/plugin-manifest.md

```diff
@@ -0,0 +1,198 @@
+    # 'files' lists which files should be extracted out from downloaded archive
+    files:
+    - from: "kubectl-restart-*/restart.sh"
+      to: "./restart.sh"
+    - from: "LICENSE"
+      to: "."
+    # 'bin' specifies the path to the the plugin executable among extracted files
+    bin: restart.sh
+```
+
+#### Documentation fields
+
+- `shortDescription:` (required): Punchline for your plugin. It should not exceed
+  a few words (to be properly shown in `kubectl krew search` output). **Avoid**
+  using redundant words.
+
+- `description:` (required): Explain what your plugin does at a high-level to help users
+  choose whether they should install the plugin. **Avoid** including list of
+  commands/options for your plugin, implement `-h`/`--help` in your plugin
+  instead.
+
+- `caveats:` If your plugin needs an extra step to work, for example, if you
+  need certain programs to be installed on user’s system, list them here.
+  **Avoid** using this field as documentation field.
+
+  `caveats` are shown to the user after installing the plugin for the first time.
+
+#### Specifying plugin download options
+
+Krew plugins must be packaged as `.zip` or `.tar.gz` archives, and should
+accessible to download from user’s machine. The relevant fields are:
+
+- `uri`: URL to the archive file (`.zip` or `.tar.gz`)
+- `sha256`: sha256 sum of the archive file
+
+```yaml
+  platforms:
+  - uri: https://github.com/foo/bar/archive/v1.2.3.zip
+    sha256: "29C9C411AF879AB85049344B81B8E8A9FBC1D657D493694E2783A2D0DB240775"
+    ...
+```
+
+#### Specifying platform-specific instructions
+
+Krew makes it possible to install the same plugin on different operating systems
+(like `windows`, `darwin` (macOS), and `linux`) and different architectures
+(like `amd64`, `386`, `arm`).
+
+To support multiple platforms, you may need to define multiple `platforms` in
+the plugin manifest. The `selector` field matches to operating systems and
+architectures using the keys `os` and `arch` respectively.
+
+**Example:** Match to a Linux or macOS platform, any architecture:
+
+```yaml
+...
+  platforms:
+  - selector:
+      matchExpressions:
+      - {key: "os", operator: "In", values: [darwin, linux]}
+```
+
+**Example:** Match to Windows 64-bit:
+
+```yaml
+  platforms:
+  - selector:
+      matchLabels:
+        os: windows
+        arch: amd64
+```
+
+**Example:** Match to Linux (any architecture):
+
+```yaml
+  platforms:
+  - selector:
+      matchLabels:
+        os: linux
+```
+
+The possible values for `os` and `arch`  come from the Go runtime. Run
+`go tool dist list` to see all possible platforms and architectures.
+
+#### Specifying files to install
+
+Each operating system may require a different set of files from the archive to
+be installed.
+
+You can use the `files` field to specify
+which files should be copied into the plugin directory while extracting the
+files out of archive.
+
+The `files:` list specifies the copy operations (like `mv(1) <from> <to>`) to
```

Comment:
Just a nit, but I think this easily confuses novice users.
```suggestion
The `files:` list specifies the copy operations (like `mv <from> <to>`) to
```

Reply:
This is from existing docs.